src/os/ObjectStore.h

```diff
@@ -1716,71 +1716,45 @@ class ObjectStore {
     static void generate_test_instances(list<Transaction*>& o);
   };
 
-  struct C_DeleteTransaction : public Context {
-    ObjectStore::Transaction *t;
-    C_DeleteTransaction(ObjectStore::Transaction *tt) : t(tt) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-  template<class T>
-  struct C_DeleteTransactionHolder : public Context {
-    ObjectStore::Transaction *t;
-    T obj;
-    C_DeleteTransactionHolder(ObjectStore::Transaction *tt, T &obj) :
-      t(tt), obj(obj) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-
   // synchronous wrappers
   unsigned apply_transaction(Sequencer *osr, Transaction& t, Context *ondisk=0) {
-    list<Transaction*> tls;
-    tls.push_back(&t);
+    vector<Transaction> tls;
```

Comment:
The ability to reuse a Transaction after a move got me wondering whether that would actually work with the default compiler-generated move operations. I wrote a unit test to verify that a Transaction is `empty()` after a move, and found that it failed.

I opened a separate pull request at https://github.com/ceph/ceph/pull/7303 that includes this unit test, along with custom move operations for struct TransactionData that allow the test to pass.